prusti-viper/src/encoder/builtin_encoder.rs

```diff
@@ -20,7 +21,14 @@ pub enum BuiltinFunctionKind {
     Unreachable(vir::Type),
     /// type
     Undefined(vir::Type),
+    /// array lookup pure function, e.g. Array$4$u32$lookup_pure
+    ArrayLookupPure {
+        array_ty: vir::Type,
+        array_len: usize,
```

Comment:
Is `array_ty` the type of the array or the type of the elements? In the former case I would expected the array length to be already part of the string contained in `array_ty`.

Reply:
renaming to `array_elem_ty` for clarity.